src/workerd/jsg/modules.h

```diff
@@ -276,19 +305,28 @@ class ModuleRegistryImpl final: public ModuleRegistry {
     const Entry& keyForRow(const Entry& row) const { return row; }
 
     bool matches(const Entry& entry, const Entry& other) const {
-      return entry.info.hash == other.info.hash;
+      return hashCode(entry) == hashCode(other);
     }
 
     bool matches(const Entry& entry, v8::Local<v8::Module>& module) const {
-      return entry.info.hash == module->GetIdentityHash();
+      return entry.info.template is<ModuleInfo>() &&
+          entry.info.template get<ModuleInfo>().hash == module->GetIdentityHash();
     }
 
     uint hashCode(v8::Local<v8::Module>& module) const {
       return kj::hashCode(module->GetIdentityHash());
     }
 
     uint hashCode(const Entry& entry) const {
-      return entry.info.hash;
+      KJ_SWITCH_ONEOF(entry.info) {
+        KJ_CASE_ONEOF(moduleInfo, ModuleInfo) {
+          return moduleInfo.hash;
+        }
+        KJ_CASE_ONEOF(src, kj::ArrayPtr<const char>) {
+          return kj::hashCode(src);
```

Comment:
I guess this is hashing the whole source code, which sounds kind of expensive. Maybe we could key off a pointer to the code instead? This always points to a static constant, right?